packages/frontend/src/state/lp/hooks.ts

```diff
@@ -13,6 +13,61 @@ import { ethers } from 'ethers'
 import { useCallback } from 'react'
 import { useGetDebtAmount, useGetTwapSqueethPrice, useGetVault } from '../controller/hooks'
 
+/*** CONSTANTS ***/
+const TICK_SPACE = 60
+const COLLAT_RATIO = 1.5
+
+/*** ACTIONS ***/
+
+// Opening a mint and LP position and depositing
+export const useOpenPositionDeposit = () => {
+  const { squeethPool } = useAtomValue(addressesAtom)
+  const address = useAtomValue(addressAtom)
+  const contract = useAtomValue(controllerHelperHelperContractAtom)
+  const handleTransaction = useHandleTransaction()
+  const getTwapSqueethPrice = useGetTwapSqueethPrice()
+  const getDebtAmount = useGetDebtAmount()
+  const openPositionDeposit = useAppCallback(
+    async (squeethToMint: BigNumber, lowerTickInput: number, upperTickInput: number, onTxConfirmed?: () => void) => {
+      if (!contract || !address) return null
+      
+      const squeethPrice = await getTwapSqueethPrice()
+      const mintWSqueethAmount = fromTokenAmount(squeethToMint, OSQUEETH_DECIMALS)
+      const ethDebt = await getDebtAmount(mintWSqueethAmount)
```

Comment:
We can combine these promises using Promise.all.

Reply:
Updated!